swap/src/database/eigensync.rs

```diff
@@ -0,0 +1,386 @@
+
+#[derive(Debug, Clone, Eq, Hash, PartialEq, Serialize, Deserialize)]
+struct InnerPeerAddressesKey(PeerId, Multiaddr);
+
+type PeerAddressesKey = KeyWrapper<InnerPeerAddressesKey>;
+
+#[derive(Debug, Clone, Reconcile, Hydrate, PartialEq, Default)]
+struct EigensyncWire {
+    states: HashMap<StateKey, String>,
+    // encode (peer_id, addr) -> "peer_id|addr", unit value as bool true
+    peer_addresses: HashMap<PeerAddressesKey, bool>,
+    // swap_id -> peer_id
+    peers: HashMap<String, String>,
+    // encode (swap_id, address?) -> "swap_id|address_or_-"; store (Decimal, String) as (String, String)
+    monero_addresses: HashMap<MoneroAddressKey, String>,
+    // buffered_transfer_proofs table
+    buffered_transfer_proofs: HashMap<String, String>, // (swap_id, proof)
+}
+
+impl From<&EigensyncDocument> for EigensyncWire {
+    fn from(src: &EigensyncDocument) -> Self {
+        let peer_addresses = src.peer_addresses.iter().map(|(key, _)|
+            (KeyWrapper::new(key.clone()), true))
+            .collect();
+
+        let monero_addresses = src.monero_addresses.iter().map(|(key, value)| {
+            (KeyWrapper::new(key.clone()), serde_json::to_string(value).unwrap())
+        }).collect();
+
+        let states = src.states.iter().map(|(key, state)| {
+            let state_json = serde_json::to_string(&Swap::from(state.clone())).unwrap();
+            (KeyWrapper::new(key.clone()), state_json)
+        }).collect();
+
+        let peers = src.peers.iter().map(|(k, v)| (k.to_string(), v.to_string())).collect();
+
+        let buffered_transfer_proofs = src.buffered_transfer_proofs.iter().map(|(k, v)| (k.to_string(), serde_json::to_string(&v).unwrap())).collect();
+
+        EigensyncWire {
+            states,
+            peer_addresses,
+            peers,
+            monero_addresses,
+            buffered_transfer_proofs,
+        }
+    }
+}
+
+impl TryFrom<EigensyncWire> for EigensyncDocument {
+    type Error = anyhow::Error;
+    fn try_from(w: EigensyncWire) -> anyhow::Result<Self> {
+        let peer_addresses = w.peer_addresses.into_iter().map(|(k, _)| {
+            let (peer_id, addr) = (k.0.0, k.0.1);
+            Ok((InnerPeerAddressesKey(peer_id, addr), ()))
+        }).collect::<anyhow::Result<HashMap<InnerPeerAddressesKey, ()>>>()?;
+
+        let monero_addresses = w.monero_addresses.into_iter().map(|(k, v)| {
+            let value: MoneroAddressValue = serde_json::from_str(&v)?;
+            Ok((k.0, value))
+        }).collect::<anyhow::Result<HashMap<InnerMoneroAddressKey, MoneroAddressValue>>>()?;
+
+        let states = w
+            .states
+            .into_iter()
+            .map(|(k, v)| {
+                let swap_id = k.0.0;
+                let timestamp = k.0.1;
+                let swap: Swap = serde_json::from_str(&v)?;
+                let state: State = swap.into();
+                // convert to utc date time from string like "2025-07-28 15:23:12.0 +00"
```

Comment:
this comments are not up to date anymore?

we aren't using timestamps like this anymore